Sources/Widgets/Representations/ArrowHandleRepresentation/index.js

```diff
@@ -152,6 +173,28 @@ function vtkArrowHandleRepresentation(publicAPI, model) {
       typedArray.points[i * 3 + 1] = coord[1];
       typedArray.points[i * 3 + 2] = coord[2];
 
+      let scale3 = state.getScale3 ? state.getScale3() : [1, 1, 1];
+      scale3 = scale3.map((x) => (x === 0 ? 2 * model.defaultScale : 2 * x));
+
+      const viewMatrix = mat3.create();
+      mat3.fromMat4(viewMatrix, model.viewMatrix);
+      const viewMatrixInv = mat3.create();
+      mat3.invert(viewMatrixInv, viewMatrix);
+
+      const displayOrientation = vec3.create();
+      vec3.transformMat3(displayOrientation, model.orientation, viewMatrixInv);
+      displayOrientation[2] = 0;
+      const displayMatrix = vtkMatrixBuilder
+        .buildFromDegree()
+        .rotateFromDirections([0, 1, 0], displayOrientation)
+        .getMatrix();
+      const displayRotation = mat3.create();
+      mat3.fromMat4(displayRotation, displayMatrix);
+      const rotation = mat3.create();
+      mat3.multiply(rotation, viewMatrix, displayRotation);
+      vec3.transformMat3(scale3, scale3, rotation);
+
+      typedArray.direction.set(rotation, 9 * i);
       typedArray.scale[i] =
         scaleFactor *
         (!state.isVisible || state.isVisible() ? 1 : 0) *
```

Comment:
No need to take care of "visibility" in requestData, just in `updateActorVisibility` should be enough

Reply:
this code is not from me (this mecanism is inherited from spherHandleRep the blueprint i used to create this file). The issue here is to link the visibility of the points directly to the state of the widget (to my mind). It allows the widget state to override every other visibility change (e.g handleVisibility from abstractWidget) by operating directly on the scale of an object.

Reply:
Does it still work without this line ?

Reply:
It doesn't. This isVisible variable is used by the state tree to display handles when user adds it to the scene. Since I opted for static handles, the process is to generate a widget when the VTK scene loads, hide the handles to the VTK scene, so we have handles loaded but not positionned yet, and then display them to the scene (here allow them to have a scale different from 0) when the user grabs focus and clicks on the scene. The problem with this line is that it is common to all the handleRepresentation used by lineWidget. It wouldn't make sense to remove it here and leave it in sphereHandle IMO. The last part however would most likely be a breaking change for other widgets.